docs/cli.md

```diff
@@ -58,12 +59,19 @@ Options:
   --num-dp-cores=<num>  Number of data plane cores [default: 4].
   --num-cp-cores=<num>  Number of control plane cores [default: 1].
   --pool=<pool>         Pool name: either infra, controlplane or dataplane.
+  --cp-mode=<mode>      Control plane core allocation mode. Possible modes:
+                        packed and spread [default: packed].
+  --dp-mode=<mode>      Data plane core allocation mode. Possible modes:
+                        packed and spread [default: packed].
   --publish             Whether to publish reports to the Kubernetes
                         API server.
   --pull-secret=<name>  Name of secret used for pulling Docker images from
                         restricted Docker registry.
   --saname=<name>       ServiceAccount name to pass
                         [default: cmk-serviceaccount].
+  --socket-id=<num>     ID of socket where allocated core should come from.
```

Comment:
It's not exactly equivalent to creating multiple pools, b/c you don't have to use it. If you don't mind use one socket or another then you don't care about this option and CMK behaviour doesn't change. If you care which socket will be used (b/c your DPDK workload needs access to specific NIC) then you can use this option.

Previous approach was simpler, but current is not so different. Most changes are in storing and gathering platform topology. Gathering those topology was required in any approach, storing was separate call from @lmdaly.